### PR TITLE
Course Change - Add the new email

### PR DESCRIPTION
**Pull request by JR-G**
## Context

Currently providers can change the details of a course at the point of offer. We'd like to let providers do this before the point of offer.

Following from [#6683](https://github.com/DFE-Digital/apply-for-teacher-training/pull/6683) this is the seventh PR in the series. This is the the email that will be sent once a provider updates a candidates course. This also includes a minor change to the content of the changed offer email.

## Changes proposed in this pull request

- Introduce a new email that the `ChangeCourse` service will send
- Update the copy of the `changed_offer` email

## Guidance to review

[New email copy](https://bat-design-history.netlify.app/manage-teacher-training-applications/emailing-a-candidate-when-their-course-details-are-changed/#changing-the-course-details-before-making-an-offer)

## Link to Trello card

https://trello.com/c/ws79kLkM

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by cwrw — COMMENTED** [review]
looking good, mostly content tweaks

File: app/views/candidate_mailer/change_course.text.erb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+The new details are:
+
+^ Training provider: <%= @current_course_option.course.provider.name %>
+^ Course: <%= @current_course_option.course.name %>
+^ Location: <%= @current_course_option.site.name %>
+^ Full time or part time: <%= @current_course_option.course.study_mode.humanize %>
+
+If you were not expecting this change, contact <%= @course_option.course.provider.name %> to find out more.
+
+# Get support
```
[the remark this answers is not in the document]
Comment (JR-G) [reply app/views/candidate_mailer/change_course.text.erb:14]: 🆗

File: app/views/candidate_mailer/changed_offer.text.erb [hunk 2]
```diff
@@ -1,13 +1,14 @@
 Dear <%= @application_form.first_name %>
 
-Your offer from <%= @course_option.course.provider.name %> to study <%= @course_option.course.name %> has been changed.
+Your offer from <%= @course_option.course.provider.name %> to study <%= @course_option.course.name_and_code %> has been changed.
```
Comment (cwrw) [inline app/views/candidate_mailer/changed_offer.text.erb:3]:
```suggestion
The offer from <%= @course_option.course.provider.name %> to study <%= @course_option.course.name_and_code %> has been changed.
```

**Review by cwrw — APPROVED** [review]

File: app/views/candidate_mailer/changed_offer.text.erb [hunk 3]
```diff
@@ -37,4 +38,6 @@ Sign in to your account to respond:
 
 <%= candidate_magic_link(@application_choice.application_form.candidate) %>
 
+Contact <%= @course_option.course.provider.name %> directly if you have any questions about this
```
[the remark this answers is not in the document]
Comment (JR-G) [reply app/views/candidate_mailer/changed_offer.text.erb:41]: 👌

**Pull request merged**